### PR TITLE
Add support for computed property names

### PR DESCRIPTION
**Pull request by rhendric**

### CODE REVIEW

**Comment by rhendric** [comment]
After this and #98, the remaining PRs I intend to submit (adding support for rest parameters, default arguments, object/array destructuring in parameters, generators/`yield`, object literal methods, classes, and the `super` keyword) start to refactor existing AST constructors, instead of merely adding new ones. If you want to make a backwards-compatible-modulo-exhaustiveness-checks release (to be followed by a backwards-incompatible release shortly thereafter), the time to do that is after this and #98 are merged. If you don't think it matters, I have no issue until waiting until the rest of the work is submitted and merged too.

**Review by erikd — APPROVED** [review]

**Pull request merged**
